src/Akeneo/Pim/Enrichment/Component/Product/Job/ComputeFamilyVariantStructureChangesTasklet.php

```diff
@@ -229,12 +206,27 @@ private function validateProducts(array $products): void
 
             if ($violations->count() !== 0) {
                 throw new \LogicException(
-                    sprintf(
-                        'Validation error for Product with identifier "%s" during family variant structure change',
-                        $product->getIdentifier()
+                    $this->buildViolationsMessage(
+                        sprintf(
+                            'One or more validation errors occured for Product with identifier "%s" during family variant structure change:',
+                            $product->getIdentifier()
+                        ),
+                        $violations
                     )
                 );
             }
         }
     }
+
+    private function buildViolationsMessage(
```

Comment:
It's strange to call a function named buildViolationsMessage in throw no ?

```suggestion
    private function buildErrorMessage(
```